spring-beans/src/main/java/org/springframework/beans/factory/support/AbstractAutowireCapableBeanFactory.java

```diff
@@ -1352,6 +1352,10 @@ protected BeanWrapper autowireConstructor(
 	 */
 	@SuppressWarnings("deprecation")  // for postProcessPropertyValues
 	protected void populateBean(String beanName, RootBeanDefinition mbd, @Nullable BeanWrapper bw) {
+		// record has not set methods
+		if (mbd.getBeanClass().isRecord()) {
```

Comment:
This check is insufficient, since the bean definition may not have a "bean class".

In any case, I will address this after merging this PR.

So there is no need to modify this PR.